File: java/DuplicateInteger.java

/*
Duplicate Integer

Given an integer array nums, return true if any value appears more than once
in the array, otherwise return false.

Input: nums = [1, 2, 3, 3]
Output: true

 */
import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean hasDuplicate(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int num : nums) {
            if (!set.add(num)) {
                return true;
            }
        }
        return false;
    }
}

File: javascript/DuplicateInteger.js
/* 
Duplicate Integer
Given an integer array nums, return true if any value appears more than once in the array, 
otherwise return false.


Input: nums = [1, 2, 3, 3]
Output: true

*/

class Solution {
  /**
   * @param {number[]} nums
   * @return {boolean}
   */
  hasDuplicate(nums) {
    return new Set(nums).size !== nums.length;
  }
}

class Solution {
  
  hasDuplicate(nums) {
    const numsSet = new Set();
    for (const num of nums) {
      if (numsSet.has(num)) return true;

      numsSet.add(num);
    }

    return false;
  }
}